datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -597,22 +640,36 @@ mod test {
     fn id_array_visitor() -> Result<()> {
         let expr = binary_expr(
             binary_expr(
-                sum(binary_expr(col("a"), Operator::Plus, lit("1"))),
+                sum(binary_expr(col("a"), Operator::Plus, lit(1))),
```

Comment:
I agree this test doesn't make sense as coercion should have happened before this pass